mitodl/bootcamp-ecommerce
# Parse WebhookRequests and import/sync FluidReview users

#### What are the relevant tickets?
Closes #137

#### What's this PR do?
- Parse a `WebhookRequest` body into separate fields
  - Added the separate fields
  - Added a status field to track if the parsing was successful or not
- If there is no existing user with the email contained in the `WebhookRequest`, create it.
- Create or update the user profile, populating the new field `fluidreview_id` with the FluidReview user id.
- Added a django command `importfluidusers` to do a batch import of FluidReview users
- Modified the python socialauth settings to associate users by email address.

Also a couple minor enhancements:
- Added `OAuthToken` to admin page
- Moved `python manage.py collectstatic` from `Dockerfile` to `docker-compose.yml`

#### How should this be manually tested?
- Send a webhook request containing valid JSON including an existing FluidReview user_id.  A `User` and `Profile` should be created in bootcamp (if not already there), and the `Profile.fluidreview_id` field should be populated.  The `WebhookRequest`'s fields should be correctly populated and the status should be set to `Succeeded`
- Send a webhook request containing invalid JSON or plain text.  The `WebhookRequest.body` field should be populated and the status should be `Failed`, other fields should be null.
- Run the command `docker-compose run web python manage.py importfluidusers` to create `User` and `Profile` objects for every FluidReview user.
- Create an account in Edx for one of the FluidReview user emails that has already been imported into bootcamp via Webhook or batch import.  Then log in to bootcamp via Edx with that account.  In another browser, log into `../admin/social_django/usersocialauth/` and verify that there is a `UserSocialAuth` object associated with the same user that was created from FluidReview.
